extensions/amp-story/1.0/amp-story.js

```diff
@@ -1404,7 +1404,7 @@ export class AmpStory extends AMP.BaseElement {
 
     // Step out if trying to navigate to the currently active page.
     if (this.activePage_ && this.activePage_.element.id === targetPageId) {
-      return Promise.resolve();
+      return this.activePage_.rewindAllMedia();
```

Comment:
There is more to rewinding a page than just rewinding its media. Having this code here might have side effects as well.
If this is trying to solve the replay case, there's a `replay_()` method where you could set the page to `PageState.NOT_ACTIVE` and back to `PLAYING`, which will restart media, page advancement (time or media based), analytics, etc.